### PR TITLE
Add setting that uses winpty for tasks

### LINKED ISSUE #73351: Watch tasks not working on Windows version 1903 with terminal.integrated.windowsEnableConpty set to true
Steps to reproduce:

- clone https://github.com/Microsoft/vscode-eslint.git
- open a terminal
- Debug viewlet
- Launch client
- Observe that building never finishes
- Now switch the terminal to the npm: watch task

Observe: building finished. Looks like the terminal is not delivering any output as long as the view is not visible.

### PR DESCRIPTION
**Pull request by alexr00**
Setting defaults to true

Fixes #73351

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/common/terminal.ts [hunk 1]
```diff
@@ -192,6 +192,12 @@ export interface IShellLaunchConfig {
 	 * provided as nothing will be inherited from the process or any configuration.
 	 */
 	strictEnv?: boolean;
+
+	/**
+	 * Moving forward, conpty will be the default. However, there are cases where conpty is not ready
+	 * do be the default. This property will force winpty to be used, even when conpty would normally be used.
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/common/terminal.ts]: do->to

File: src/vs/workbench/contrib/tasks/electron-browser/task.contribution.ts [hunk 2]
```diff
@@ -2749,3 +2751,19 @@ ProblemMatcherRegistry.onMatcherChanged(() => {
...
 });
+
+const configurationRegistry = Registry.as<IConfigurationRegistry>(Extensions.Configuration);
+configurationRegistry.registerConfiguration({
+	id: 'Tasks',
+	order: 100,
+	title: nls.localize('tasksConfigurationTitle', "Tasks"),
+	type: 'object',
+	properties: {
+		'tasks.terminal.windowsEnableConpty': {
```
Comment (Tyriar) [inline src/vs/workbench/contrib/tasks/electron-browser/task.contribution.ts]: windowsAllowConpty?

**Review by Tyriar — APPROVED** [review]

**Comment by Tyriar** [comment]
I made my suggested changes and verified that conpty is only enabled when the setting is turned on. @RMacfarlane please review 🙂

File: src/vs/workbench/contrib/tasks/electron-browser/task.contribution.ts [hunk 3]
```diff
@@ -2749,3 +2751,19 @@ ProblemMatcherRegistry.onMatcherChanged(() => {
...
+
+const configurationRegistry = Registry.as<IConfigurationRegistry>(Extensions.Configuration);
+configurationRegistry.registerConfiguration({
+	id: 'Tasks',
+	order: 100,
+	title: nls.localize('tasksConfigurationTitle', "Tasks"),
+	type: 'object',
+	properties: {
+		'tasks.terminal.windowsAllowConpty': {
+			markdownDescription: nls.localize('tasks.terminal.windowsAllowConpty', "Works in conjunction with the terminal.integrated.windowsEnableConpty setting. Both must be enabled for tasks to use conpty. Defaults to false."),
```
Comment (RMacfarlane) [inline src/vs/workbench/contrib/tasks/electron-browser/task.contribution.ts]: I think using `#terminal.integrated.windowsEnableConpty#` would create a link in the settings editor

**Review by RMacfarlane — APPROVED** [review]

**Pull request merged**